Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -7394,7 +7996,43 @@ class BBQJIT {
         return { };
     }
 
-    PartialResult WARN_UNUSED_RETURN addCallRef(const TypeDefinition&, Vector<Value>&, ResultList&) BBQ_STUB
+    PartialResult WARN_UNUSED_RETURN addCallRef(const TypeDefinition& originalSignature, Vector<Value>& args, ResultList& results)
+    {
+        Value callee = args.takeLast();
+        const TypeDefinition& signature = originalSignature.expand();
+        ASSERT(signature.as<FunctionSignature>()->argumentCount() == args.size());
+
+        CallInformation callInfo = wasmCallingConvention().callInformationFor(signature, CallRole::Caller);
+        Checked<int32_t> calleeStackSize = WTF::roundUpToMultipleOf(stackAlignmentBytes(), callInfo.headerAndArgumentStackSizeInBytes);
+        m_maxCalleeStackSize = std::max<int>(calleeStackSize, m_maxCalleeStackSize);
+
+        GPRReg calleeInstance, calleeCode, jsCalleeAnchor;
+
+        {
+            ScratchScope<3, 0> scratches(*this);
```

Comment:
Please update this code to the latest convention. `calleeCode` always needs to be in GPRInfo::nonPreservedNonArgumentGPR1 to make shuffling work correctly. See updated `addCallIndirect` code changed in https://github.com/WebKit/WebKit/commit/76f47aa9042386307ab0e61744870833fd981a76